src/goto-programs/remove_instanceof.h

```diff
@@ -17,8 +17,12 @@ Author: Chris Smowton, chris.smowton@diffblue.com
 #include "goto_model.h"
 
 void remove_instanceof(
-  symbol_tablet &symbol_table,
-  goto_functionst &goto_functions);
+  goto_functionst::goto_functiont &function,
```

Comment:
Would you mind explaining your thinking behind this suggestion? No particular objections but don't want to make changes for no reason.

Reply:
You are only using a single member of the `goto_functiont` class (the `body`), hence this function seems applicable in all contexts where the potential caller holds a `goto_programt`. It thus makes the function more general. (It may be worth scrutinising the code base for all uses of `goto_functiont` as really only those cases where type information is required should be using that.

Reply:
A quick search reveals the following methods that look to me like passes using `goto_functiont`:
`remove_complex`, `remove_vector`, `remove_java_new`, `adjust_float_expressions`, `rewrite_union`, `remove_asm`
Making `remove_instanceof` not use it when these do seems like a bad idea. If you want to change them all then feel free to create an issue.

Reply:
As far as I am aware the code base only passes goto_programt for manipulations of goto programs.

Reply:
... which is what all these functions do. Added as #1681.